core/services/directrequest/delegate.go

```diff
@@ -232,29 +232,34 @@ func (l *listener) processCancelOracleRequests() {
 
 func (l *listener) handleReceivedLogs(mailbox *utils.Mailbox[log.Broadcast]) {
 	for {
+		select {
+		case <-l.chStop:
+			return
```

Comment:
It is arguably a correctness problem that it wouldn't yield soon enoug and could slow shutdown, but was not as significant of an effect before this fix because errors elsewhere from other systems being down would exit the loop.